File: src/types/enums.ts
export const RoomStatus = {
  lobby: 'lobby',
  waiting: 'waiting',
  submissions_open: 'submissions_open',
  submissions_closed: 'submissions_closed',
  guessing: 'guessing',
  revealed: 'revealed',
  completed: 'completed',
  ended: 'ended',
} as const;

export type RoomStatus = (typeof RoomStatus)[keyof typeof RoomStatus];

export const RoundStatus = {
  waiting: 'waiting',
  submissions_open: 'submissions_open',
  submissions_closed: 'submissions_closed',
  guessing: 'guessing',
  revealed: 'revealed',
  completed: 'completed',
} as const;

export type RoundStatus = (typeof RoundStatus)[keyof typeof RoundStatus];

export const ParticipantRole = {
  owner: 'owner',
  player: 'player',
} as const;

export type ParticipantRole = (typeof ParticipantRole)[keyof typeof ParticipantRole];

File: app/models/enums.py
import enum

class RoomStatus(str, enum.Enum):
    lobby = "lobby"
    waiting = "waiting"
    submissions_open = "submissions_open"
    submissions_closed = "submissions_closed"
    guessing = "guessing"
    revealed = "revealed"
    completed = "completed"
    ended = "ended"

class RoundStatus(str, enum.Enum):
    waiting = "waiting"
    submissions_open = "submissions_open"
    submissions_closed = "submissions_closed"
    guessing = "guessing"
    revealed = "revealed"
    completed = "completed"

class ParticipantRole(str, enum.Enum):
    owner = "owner"
    player = "player"
